account_multicurrency_revaluation/model/res_currency.py

```diff
@@ -21,32 +21,30 @@
 
 import time
 
-from osv import osv
-from tools.translate import _
+from openerp import models, api, _
+from openerp.exceptions import Warning
 
 
-class res_currency(osv.osv):
+class ResCurrency(models.Model):
 
     _inherit = 'res.currency'
 
-    def _get_conversion_rate(self, cr, uid, from_currency,
-                             to_currency, context=None):
-        if context is None:
-            context = {}
+    @api.model
+    def _get_conversion_rate(self, from_currency, to_currency):
+        context = self._context or {}
         if 'revaluation' in context:
-            currency = self.browse(cr, uid, from_currency.id, context=context)
+            currency = self.with_context(context).browse(from_currency.id)
```

Comment:
And BTW 

currency = self.browse(from_currency.id)
is simply
currency = from_currency